### PR TITLE
TINY-6311 Improved cell selection css to achieve a semi-transparent overlay

### PR DESCRIPTION
**Pull request by lostkeys**
Related Ticket: TINY-6311

Description of Changes:
Table cell selection is now done using a semi-transparent pseudo element on top of the cell instead of manipulating the cell background directly. This allows cell content to shine through. Tested in IE11. Compare this before/after image:

image
Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/oxide/src/less/theme/content/selection/selection.less [hunk 1]
```diff
@@ -83,6 +83,19 @@
     }
   }
 
+  // Apply cell selection as a semi-transparent overlay instead of manipulating the cell background itself
+  td[data-mce-selected]::after,
+  th[data-mce-selected]::after {
+    background-color: @table-selection-color;
```
Comment (ashfordneil) [inline modules/oxide/src/less/theme/content/selection/selection.less:89]: Do we normally write CSS properties in alphabetical order? If that's a convention then that's fine, but if not could we group them by similar properties — put bottom / top / left / right all together, background-colour / opacity, etc..

Comment (lnewson) [reply modules/oxide/src/less/theme/content/selection/selection.less:89]: If you look at the stylelint rules you'll see it's required to write them in alphabetical order.

**Comment by lostkeys** [comment]
Readability might be an issue, although I expect users be more focused on the selection than on what's actually selected. The previous solution was actually destructive in the way it removed the background colors on selected cells, potentially loosing context.

That said, I've added a color multiplier property to the CSS, making black shine through att full strength which seems to work in my testing. It won't work on IE11 and older versions of Edge but I gather that if you're concerned with accessibility, you're not using ie11 anyway.

This is how it looks now:
Screen Shot 2020-08-18 at 06 51 43

**Comment by lostkeys** [comment]
I've spent some time improving especially IE11. I feel this is acceptable given many of our peers look the same.

IE11 below
IE11

All others
All others

File: modules/oxide/src/less/theme/content/selection/selection.less [hunk 2]
```diff
@@ -69,20 +71,42 @@
   // Selection of whole table cells
   td[data-mce-selected],
   th[data-mce-selected] {
-    background-color: @table-selection-color !important;
+    outline: none; //Prevent selection to be visible
+    position: relative;
+    user-select: none; //Prevent selection to be visible
```
Comment (lnewson) [inline modules/oxide/src/less/theme/content/selection/selection.less:76]: This is quite likely going to break things, as the content needs to be selectable. I suspect this is why some tests are failing.

**Review by lnewson — CHANGES_REQUESTED** [review]

File: modules/oxide/src/less/theme/content/selection/selection.less (hunk 2 above)
Comment (lostkeys) [reply modules/oxide/src/less/theme/content/selection/selection.less]: Removed

**Review by lnewson — APPROVED** [review]

**Pull request merged**